Gisaid/Genbank download tweaks

ehoops-cz opened the pull request:
### Summary:
- **What:** `Update endpoint usage`
- **Ticket:** none
- **Env:** https://template-download-frontend.dev.czgenepi.org/

To get fasta file, use the `ORG_API.SAMPLES_FASTA_DOWNLOAD` endpoint and put the public repository in the request body:
```
{
        "sample_ids": [sample.public_identifier],
        "public_repository_name": "GISAID",
}
```
To get submission template, use the new `/v2/orgs/{group.id}/samples/submission_template` endpoint and pass:
```
   {
        "sample_ids": [sample.public_identifier for sample in samples],
        "date": today.strftime("%Y-%m-%d"),
        "page": "0",
        "public_repository_name": "GenBank",
    }
```
Use the "content-disposition" header to get the filename.

### Demos:

### Notes:
I added `resp.headers["Access-Control-Expose-Headers"] = "Content-Disposition"` to the submission_template endpoint for the filename to work.  Extra eyes on this would be great since I was just being a sheep and following what i thought should be there...

### Checklist:
- [x] I merged latest `<base branch>`
- [x] I manually verified the change
- [x] I added labels to my PR
- [ ] I tested in multiple browsers
- [ ] I added relevant unit tests
- [ ] I have notified others of changes they need to make locally (migrations, jobs, package updates, etc)

ehoops-cz committed:
wip

ehoops-cz committed:
Merge remote-tracking branch 'origin/trunk' into g-g-download-mod

ehoops-cz committed:
Use filename from content-disposition header

ehoops-cz committed:
Adjust api calls for fasta and templates. Use multiple calls if more than 999 samples.

ehoops-cz committed:
Add header to tsv_streamer response.

ehoops-cz commented on .pre-commit-config.yaml:
```diff
@@ -9,7 +9,7 @@ repos:
         types: [file, python]
   - repo: https://github.com/psf/black
-    rev: 21.9b0
+    rev: 22.3.0 
```
I was getting an error running the pre-commit hook.  [This fixed it](https://stackoverflow.com/questions/71673404/importerror-cannot-import-name-unicodefun-from-click).  I'm not sure how else I should be testing this though.

phoenixAja commented on src/frontend/src/common/components/library/data_subview/components/DownloadModal/index.tsx:
```diff
@@ -358,24 +360,43 @@ const DownloadModal = ({
+            endpoint: ORG_API.SAMPLES_TEMPLATE_DOWNLOAD,
+            page,
+            publicRepositoryName: PUBLIC_REPOSITORY_NAME.GISAID,
+            sampleIds: checkedSampleIds,
```
checkedSamples needs to be subsetted to go along with the page, so for page0 we need checkedSamples[0-999]

phoenixAja replied:
unless i'm missing something

lvreynoso replied:
0-998, right? For 999 samples

phoenixAja replied:
was it 999 or 1000? I'm having a hard time finding the supporting documentation for it. @danrlu can you confirm please?

danrlu replied:
the limit is 1000, so per our last discussion ppl think we should play safe and just do 999 based on my memory, or I failed to communicate clearly XD Let me quickly double check and get back to you.

phoenixAja replied:
thanks dan! let's keep it at 999 then just in case!

ehoops-cz replied:
This other ticket also uses 999: https://app.shortcut.com/genepi/story/194965/as-a-user-if-i-select-more-than-999-samples-i-want-to-be-notified-that-my-template-files-will-be-chopped-up

lvreynoso commented:
👍

phoenixAja reviewed (changes requested):
i think we need to update the checkedSamples that are sent in to the request, besides that it looks good!

ehoops-cz committed:
Send subset of sampleIds

ehoops-cz committed:
Remove chunking.

ehoops-cz committed:
Use GISAID User Name instead of Submitter ID. Filter to only send completed samples to the backend.

ehoops-cz committed:
Remove lint.

ehoops-cz commented:
Removed chunking from the frontend including the multiple file callout.
Renamed GISAID Submitter ID to GISAID User Name.
Filtered to only send completedSampleIds and not failed ones.
Updated Failed sample callout copy.

ehoops-cz committed:
Update copy.
